extensions/smallrye-graphql/deployment/src/main/java/io/quarkus/smallrye/graphql/deployment/SmallRyeGraphQLProcessor.java

```diff
@@ -158,20 +162,48 @@ void registerNativeImageResources(BuildProducer<ServiceProviderBuildItem> servic
                 .produce(ServiceProviderBuildItem.allProvidersFromClassPath(SmallRyeGraphQLConfigMapping.class.getName()));
     }
 
+    @BuildStep
+    SmallRyeGraphQLIndexBuildItem createIndex(TransformedClassesBuildItem transformedClassesBuildItem) {
```

Comment:
I am actually not sure we want to do this for all transformed classes... I really think it should be limited to only the classes we know are problematic for this processor, which for now is the Panache classes. My fear is that if we doing it for all classes and in various places, the build could slow down considerably.

Reply:
Actually, scratch that. I see you are using the result of the transformation instead.

This should indeed work nicely - I just hope it won't cause ordering problems in the future.

Reply:
I can also get the list of panache classes (as Strings) in a BuildItem, but using that here creates a Cycle. Let me play around some more and see what I can figure out. The other option is to use the current index (before the merge) and get all panache classes. w.d.y.t ?

Reply:
I think that what you did here is clean and simple. 
I guess let's not add any more complexity unless it actually becomes necessary. The good thing is that if this becomes a problem (which it probably won't), we can definitely do others things, so let's just keep them in mind.

Reply:
One thing however that should be additionally tested, is dev-mode. Adding and removing fields should work properly.
This solution should work, but best have a test for it

Reply:
OK cool. I can add that.

Reply:
Added "add" and "remove" field hot reload test.